### DISCUSSION TITLE
Does encryption wipe the device before setting up dm-crypt?

### DISCUSSION THREAD

**Discussion by dseomn** [rook/rook#7603] [Q&A] [answered]
https://wiki.archlinux.org/index.php/Dm-crypt/Drive_preparation says the below, and I've read similar things in a couple other places:
> Before encrypting a drive, it is recommended to perform a secure erase of the disk by overwriting the entire drive with random data. To prevent cryptographic attacks or unwanted file recovery, this data is ideally indistinguishable from data later written by dm-crypt.

When creating encrypted OSDs with Rook/Ceph, is cryptographically secure random data written to those disks/partitions before dm-crypt is set up on them? (I don't see anything in https://docs.ceph.com/en/latest/ceph-volume/lvm/encryption/ to indicate one way or the other.)

**Comment by leseb** [comment, marked as answer]
Rook never wipes the disks prior to deploy a cluster, whether the encryption is enabled or not. We leave this to the admin.
